pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -330,6 +337,20 @@ def __init__(self, training: bool, device: Optional[Union[str, torch.device]] =
         self.device: Optional[Union[str, torch.device]] = device
         self.fx_validator = FxValidator()
 
+    @property
+    def result_metrics(self) -> List[ResultMetric]:
+        o = []
+        for v in self.values():
+            if isinstance(v, ResultMetric):
+                o.append(v)
+            elif isinstance(v, ResultCollection):
+
+                def append_fn(v: ResultMetric) -> None:
+                    o.append(v)
+
+                apply_to_collection(v, ResultMetric, append_fn)
```

Comment:
you could probably do the `apply_to_collection` over `self.values()` directly, or not?

Reply:
Yes.